woocommerce/pinterest-for-woocommerce
# Pinterest Conversions API

### Changes proposed in this Pull Request:

Closes #771 .

Adds Conversions API implementation as well as Pinterest Tag implementation refactoring. Now we have a general Tracking class which hooks into events of interest and had trackers attacked to handle action accordingly. 

e.g. Tracking has two tracker implementations: Tag and Conversions. Tracking class hooks into `add to cart` event and when action is fired it notifies trackers about the event so the last could perform their corresponding actions.

Conversions API events are logged under `pinterest-for-woocommerce-conversions-<date>` log files so the user could see what was logged.

<img width="1447" alt="WooCommerce_status_‹_WordPress_Pinterest_—_WordPress" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/99bdf5f0-f41a-4b19-9368-8a92f955ef8f">

Pinterest has events log which shows events receive through Tag and Conversion APIs. It is located under Ads -> Conversions

<img width="1447" alt="_1__Pinterest" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/8157728a-2010-410a-9ba2-edb188aade1d">

<img width="1091" alt="Cursor_and__1__Pinterest" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/4c52c745-1b87-4879-a57c-f7da50d5620f">

Unfortunately I was not able to find where Pinterest stores any event details we report. Could be it is not yet implemented or something.

### Detailed test instructions:

1. Checkout `add/pinterest-conversions` branch
2. Connect to your Pinterest Account.
3. Navigate the website, visit product pages, perform product searches, add to cart and checkout events.
4. Check WooCommerce conversion logs and Pinterest Tag Helper while navigating the website (logs will show you data reported through Conversion APIs, Pinterest Tag Helper will show you data reported through it).
5. Visit Pinterest Dashboard, the page under Ads > Conversions to see events reported.

### Additional details:


### Changelog entry

> Add - Pinterest Conversions API implementation.
